lib/matplotlib/tests/test_axes.py

```diff
@@ -3992,6 +3992,59 @@ def test_hlines():
     ax5.set_ylim(0, 15)
 
 
+def generate_lines_with_colors_inputs():
+    colors = ['red', 'green', 'blue', 'purple', 'orange']
+    xy_nan = [1, 2, 3, np.nan, 5]
+    xy_mask = np.ma.masked_equal([1, 2, 3, 4, 5], 4)
+    lines_nan = [{'base_xy': xy_nan,
+                    'base_max': np.ones(5),
+                    'base_min': [0],
+                    'linewidth': 5,
+                    'colors': colors}]
+    lines_mask = [{'base_xy': xy_mask,
+                    'base_max': np.ones(5),
+                    'base_min': [0],
+                    'linewidth': 5,
+                    'colors': colors}]
+
+    return [*lines_nan, *lines_mask]
+
+
+@pytest.mark.parametrize('kwargs', generate_lines_with_colors_inputs())
+@check_figures_equal(extensions=["png"])
+def test_vlines_with_colors(fig_test, fig_ref, kwargs):
```

Comment:
You can greatily simplify the tests:

- The only variable part is the data. All other parameters are identical. You can just hard-code them. No need to build dicts and pass them as `kwargs`.
- ymin/ymax can be scalars.
- You can test `hlines` and `vlines` within one figure: 
![image](https://user-images.githubusercontent.com/2836374/77075043-067bcc00-69f2-11ea-8a74-07fba1b864fd.png)

Reply:
- Removed passing kwargs, hard-code the identical values. 
- ymin/ymax are scalars now
- hlines and vlines is in one figure with 2 subplots now.